cli/fathom_web/commands/vectorize.py

```diff
@@ -0,0 +1,340 @@
+    FATHOM_FOX_DIR: Path to the FathomFox source directory
+    FATHOM_TRAINEES_DIR: Path to the Fathom Trainees source directory
+
+    \b
+    This tool will run an instance of Firefox to use the Vectorizer within the
+    FathomFox adddon. Required for this tool to work are:
+      * node
+      * yarn
+      * A FathomFox repository checkout
+      * A Fathom Trainees repository checkout
+      * A copy of Firefox
+      * A geckoview driver downloaded and accessible in your PATH environment variable
+    """
+    # TODO: Try a class based approach so I don't need these =None statements or need to pass the temp_dir around
+    firefox = None
+    firefox_pid = None
+    geckoview_pid = None
+    server = None
+    server_thread = None
+    graceful_shutdown = False
+    temp_dir = TemporaryDirectory()
+    try:
+        sample_filenames = run_fathom_list(samples_directory)
+        fathom_fox, fathom_trainees = build_fathom_addons(ruleset_file, fathom_fox_dir, fathom_trainees_dir, temp_dir)
+        server, server_thread = run_file_server(samples_directory)
+        firefox, firefox_pid, geckoview_pid = configure_firefox(fathom_fox, fathom_trainees, output_directory, show_browser, temp_dir)
+        firefox = run_vectorizer(firefox, fathom_type, sample_filenames)
+        graceful_shutdown = True
+    # TODO: How to set the exit code here?
+    except KeyboardInterrupt:
+        # Swallow the KeyboardInterrupt here so we can perform our teardown
+        # instead of letting Click do something with it.
+        pass
+    except UngracefulError as e:
+        print(f'\n\n{e}')
+    except GracefulError as e:
+        print(f'\n\n{e}')
+        graceful_shutdown = True
+    finally:
+        teardown(firefox, firefox_pid, geckoview_pid, server, server_thread, graceful_shutdown)
+
+
+def run_fathom_list(samples_directory):
+    """Use fathom-list to get a string of the filenames to run through the vectorizer."""
+    print('Running fathom-list to get list of sample filenames...', end='', flush=True)
+    # TODO: Better error message for not having fathom-list
+    result = subprocess.run(['fathom-list', samples_directory, '-r'], capture_output=True)
+    sample_filenames = result.stdout.decode()
+    print('Done')
+    return sample_filenames
+
+
+def build_fathom_addons(ruleset_file, fathom_fox_dir, fathom_trainees_dir, temp_dir):
+    """
+    Create .xpi files for fathom addons to load into Firefox.
+
+    The Firefox webdriver requires we load custom addons using .xpi files. We
+    need to load both FathomFox and Fathom Trainees. For Fathom Trainees, we
+    also need to run yarn to package up the addon with the user's ruleset.js.
+    """
+    print('Building fathom addons for Firefox...', end='', flush=True)
+    fathom_fox = create_xpi_for(pathlib.Path(fathom_fox_dir) / 'addon', 'fathom-fox', temp_dir)
+    shutil.copyfile(ruleset_file, f'{fathom_trainees_dir}/src/trainees.js')
```

Comment:
It doesn't seem that any errors are thrown for `fathom-vectorize` if the `ruleset.js` file passed in has `import` statements; although certainly when the script tries to load, it should throw an error in Firefox. Should we consider either checking in `fathom-vectorize` for `import` statements or checking in Fathom Fox and sending an error message back in this situation?

This would be a follow-up issue either way. Not blocking for this PR.

Reply:
That is a good point. I lean towards letting Fathom Fox tell us what is wrong instead of doing up front validation.

Reply:
Okay; would you mind filing a follow-up issue for this? I imagine this will be a pretty common situation, and when I tried it, the error(s) were pretty opaque (nothing showed up in the Vectorizer page).